plugin/src/main/java/io/jenkins/plugins/casc/impl/secrets/VaultSecretSource.java

```diff
@@ -27,65 +27,138 @@
+    private static final String CASC_VAULT_URL = "CASC_VAULT_URL";
+    private static final String CASC_VAULT_MOUNT = "CASC_VAULT_MOUNT";
+    private static final String CASC_VAULT_TOKEN = "CASC_VAULT_TOKEN";
+    private static final String CASC_VAULT_APPROLE = "CASC_VAULT_APPROLE";
+    private static final String CASC_VAULT_APPROLE_SECRET = "CASC_VAULT_APPROLE_SECRET";
+    private static final String CASC_VAULT_NAMESPACE = "CASC_VAULT_NAMESPACE";
+    private static final String CASC_VAULT_ENGINE_VERSION = "CASC_VAULT_ENGINE_VERSION";
+    private static final String CASC_VAULT_PATHS = "CASC_VAULT_PATHS";
+    private static final String CASC_VAULT_PATH = "CASC_VAULT_PATH"; // TODO: deprecate!
+
+    private static final String DEFAULT_ENGINE_VERSION = "2";
+    private static final String DEFAULT_USER_BACKEND = "userpass";
+
+
     public VaultSecretSource() {
-        String vaultFile = System.getenv("CASC_VAULT_FILE");
+        Optional<String> vaultFile = Optional.ofNullable(System.getenv(CASC_VAULT_FILE));
         Properties prop = new Properties();
-        if (vaultFile != null) {
-            try (FileInputStream input = new FileInputStream(vaultFile)) {
-                prop.load(input);
-                if (prop.isEmpty()) {
-                    LOGGER.log(Level.WARNING, "Vault secret file is empty");
-                }
-            } catch (IOException ex) {
-                LOGGER.log(Level.WARNING, "Failed to load Vault secrets from file", ex);
+        vaultFile.ifPresent(file -> readPropertiesFromVaultFile(file, prop));
+
+        // Parse variables
+        Optional<String> vaultPw = getVariable(CASC_VAULT_PW, prop);
+        Optional<String> vaultUser = getVariable(CASC_VAULT_USER, prop);
+        Optional<String> vaultUrl = getVariable(CASC_VAULT_URL, prop);
+        Optional<String> vaultMount = getVariable(CASC_VAULT_MOUNT, prop);
+        Optional<String> vaultToken = getVariable(CASC_VAULT_TOKEN, prop);
+        Optional<String> vaultAppRole = getVariable(CASC_VAULT_APPROLE, prop);
+        Optional<String> vaultAppRoleSecret = getVariable(CASC_VAULT_APPROLE_SECRET, prop);
+        Optional<String> vaultNamespace = getVariable(CASC_VAULT_NAMESPACE, prop);
+        Optional<String> vaultEngineVersion = getVariable(CASC_VAULT_ENGINE_VERSION, prop);
+        Optional<String[]> vaultPaths = getCommaSeparatedVariables(CASC_VAULT_PATHS, prop)
+                .map(Optional::of)
+                .orElse(getCommaSeparatedVariables(CASC_VAULT_PATH, prop)); // TODO: deprecate!
+
+        // Check mandatory variables are set
+        if(!vaultUrl.isPresent() || !vaultPaths.isPresent()) return;
+
+        // Check defaults
+        if(!vaultMount.isPresent()) vaultMount = Optional.of(DEFAULT_USER_BACKEND);
+        if(!vaultEngineVersion.isPresent()) vaultEngineVersion = Optional.of(DEFAULT_ENGINE_VERSION);
+
+        // configure vault client
+        Vault vault = null;
+        VaultConfig vaultConfig = null;
```

Comment:
Avoiding potential null pointer warning
```suggestion
        VaultConfig vaultConfig = new VaultConfig().address(vaultUrl.get());
```

Reply:
Done :+1: